suite-native/module-device-onboarding/src/hooks/useInitiateThpConnection.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectThpStep } from '@suite-common/thp';
+import {
+    DeviceOnboardingStackParamList,
+    DeviceOnboardingStackRoutes,
+    RootStackParamList,
+    StackToTabCompositeProps,
+} from '@suite-native/navigation';
+import TrezorConnect from '@trezor/connect';
+
+type NavigationProp = StackToTabCompositeProps<
+    DeviceOnboardingStackParamList,
+    DeviceOnboardingStackRoutes.ThpPairingInfo,
+    RootStackParamList
+>;
+
+export const useInitiateThpConnection = () => {
+    const navigation = useNavigation<NavigationProp>();
+
+    const thpStep = useSelector(selectThpStep);
+
+    const initiateThpConnection = () => {
+        // Device is acquired by the FW installation hook, just respond as expected.
+        TrezorConnect.uiResponse({ type: 'ui-receive_confirmation', payload: true });
+    };
+
+    useEffect(() => {
+        if (thpStep === 'ConfirmConnectionBeforePairing' || thpStep === 'ConfirmOnlyConnection') {
```

Comment:
we are duplicating the logic from [useHandleDeviceconnection](https://github.com/trezor/trezor-suite/pull/20936/files#diff-2c42ad1964a723c2be0af81a1caacafe51f7c0aaabc5bcbf253830598d96311cR120) is possible to unify it, maybe creating a dedicated global hook? The reason is whether the condition or logic would change we would need to be also aware of the other locations of this call

Reply:
No, we're not, we navigate to a different screen.